lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyCompletionParticipant.java

```diff
@@ -53,12 +53,24 @@ public void onAttributeValue(String valuePrefix, ICompletionRequest request, ICo
         Properties variableProps = SettingsService.getInstance()
                 .getVariablesForServerXml(request.getXMLDocument()
                         .getDocumentURI());
-        String variableName = valuePrefix.replace("$", "")
-                .replace("{", "")
-                .replace("}", "");
-        variableProps.entrySet().stream().filter(it -> it.getKey().toString().toLowerCase().contains(variableName.toLowerCase()))
+        // value prefix will contain all existing attribute value, till the cursor
+        String variableName = valuePrefix;
+        String replacePrefix;
+        if (valuePrefix.contains("${")) {
+            variableName = valuePrefix.substring(valuePrefix.lastIndexOf("${"))
```

Comment:
I feel like this won't work if the user entered something like `${myFirstVar}/mySecond` and triggered completion at the end of the String. This code will decide to do completion on `myFirstVar` won't it? When you demoed it, I thought it worked right, but from looking at this code I don't think so.

Reply:
for showing completion of second variable, variable should always prefix with ${ . Hence show no completion here